test/get-function-name-with-kind.js

```diff
@@ -64,6 +64,11 @@ describe("The 'getFunctionNameWithKind' function", () => {
         "class A { static async foo() {} }": "static async method 'foo'",
         "class A { static get foo() {} }": "static getter 'foo'",
         "class A { static set foo(a) {} }": "static setter 'foo'",
+        "export default async function* foo() {}":
+            "async generator function 'foo'",
+        "export default async function* () {}":
+            "async generator function 'default'",
+        "export default async () => {}": "async arrow function 'default'",
```

Comment:
It's normally the name of the function that we put in between quotes (`''`).
`default` isn't the name of the function, it's the type of export that you're doing.
The function is an anonymous function, so it has no name.

I think we should get it like 🤔

```suggestion
        "export default async function* () {}":
            "default async generator function",
        "export default async () => {}": "default async arrow function",
```

Reply:
`default` is the function name.

We do the same to 

```js
a.b = function() {} // -> `function 'b'`
```

Reply:
```console
>node
Welcome to Node.js v18.12.1.
Type ".help" for more information.
> (await import("data:text/javascript,export default function(){}")).default.name
'default'
>
```